### PR TITLE
Migrate `pkg/kubelet/pod,pleg` to structured logging

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

For Structured Logging Migration: https://github.com/kubernetes/enhancements/issues/1602
Ref: https://github.com/kubernetes/kubernetes/issues/98976
Migrate `pkg/kubelet/(pod, pleg)` to structured logging

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Migrate `pkg/kubelet/(pod, pleg)` to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/ok-to-test

**Comment by gjkim42** [comment]
It is a flake (https://github.com/kubernetes/kubernetes/issues/98893)
/retest

File: pkg/kubelet/pod/mirror_client.go [hunk 1]
```diff
@@ -119,17 +119,17 @@ func (mc *basicMirrorClient) DeleteMirrorPod(podFullName string, uid *types.UID)
...
-	klog.V(2).Infof("Deleting a mirror pod %q (uid %#v)", podFullName, uid)
+	klog.V(2).InfoS("Deleting a mirror pod", "fullName", podFullName, "uid", uid)
 	var GracePeriodSeconds int64
 	if err := mc.apiserverClient.CoreV1().Pods(namespace).Delete(context.TODO(), name, metav1.DeleteOptions{GracePeriodSeconds: &GracePeriodSeconds, Preconditions: &metav1.Preconditions{UID: uid}}); err != nil {
 		// Unfortunately, there's no generic error for failing a precondition
 		if !(apierrors.IsNotFound(err) || apierrors.IsConflict(err)) {
 			// We should return the error here, but historically this routine does
 			// not return an error unless it can't parse the pod name
-			klog.Errorf("Failed deleting a mirror pod %q: %v", podFullName, err)
+			klog.ErrorS(err, "Failed deleting a mirror pod", "fullName", podFullName)
```
Comment (ehashman) [inline pkg/kubelet/pod/mirror_client.go:132]: I'm not sold on the `fullName` label, should this just be `name`? @serathius wdyt?

**Review by ehashman — COMMENTED** [review]
Mostly LGTM!

/triage accepted

File: pkg/kubelet/pod/mirror_client.go (hunk 1 above)
Comment (gjkim42) [reply pkg/kubelet/pod/mirror_client.go:132]: In this case, there is a log above this line saying that parsing `fullName` failed, which can't be replaced by `name`. 
I prefer `fullName` for the consistency between logs.

Comment (gjkim42) [reply pkg/kubelet/pod/mirror_client.go:132]: Perhaps, I can use `klog.KRef` there is name and namespace.

**Comment by yangjunmyfm192085** [comment]
I think I have repaired some files here
https://github.com/kubernetes/kubernetes/pull/98850

File: pkg/kubelet/pod/mirror_client.go (hunk 1 above)
Comment (serathius) [reply pkg/kubelet/pod/mirror_client.go]: Using `klog.KRef` SGTM

File: pkg/kubelet/container/helpers.go [hunk 2]
```diff
@@ -82,13 +81,13 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 	}
 	// Check RestartPolicy for dead container
 	if pod.Spec.RestartPolicy == v1.RestartPolicyNever {
-		klog.V(4).Infof("Already ran container %q of pod %q, do nothing", container.Name, format.Pod(pod))
+		klog.V(4).InfoS("Already ran container, do nothing", "container", container.Name, "pod", klog.KObj(pod))
```
Comment (serathius) [inline pkg/kubelet/container/helpers.go:84]:
```suggestion
		klog.V(4).InfoS("Already ran container, do nothing", "containerName", container.Name, "pod", klog.KObj(pod))
```

File: pkg/kubelet/container/helpers.go [hunk 3]
```diff
@@ -82,13 +81,13 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
...
 		return false
 	}
 	if pod.Spec.RestartPolicy == v1.RestartPolicyOnFailure {
 		// Check the exit code.
 		if status.ExitCode == 0 {
-			klog.V(4).Infof("Already successfully ran container %q of pod %q, do nothing", container.Name, format.Pod(pod))
+			klog.V(4).InfoS("Already successfully ran container, do nothing", "container", container.Name, "pod", klog.KObj(pod))
```
Comment (serathius) [inline pkg/kubelet/container/helpers.go:90]:
```suggestion
			klog.V(4).InfoS("Already successfully ran container, do nothing", "containerName", container.Name, "pod", klog.KObj(pod))
```

**Review by serathius — CHANGES_REQUESTED** [review]

**Comment by gjkim42** [comment]
> I think I have modified some files here
> #98850

Reset `pkg/kubelet/container`

**Comment by gjkim42** [comment]
> /approve

I think the prow bot couldn't recognize the approval for some reason.

**Comment by gjkim42** [comment]
> /approve

@mrunalp
/assign @mrunalp
I think the prow bot couldn't recognize the approval for some reason. Could you approve the PR again, please?

**Pull request merged**